src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -75,51 +83,140 @@ public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string repo
-                        repoOwner,
-                        repoName,
-                        new CommitRequest { Sha = commitSha },
-                        new ApiOptions { PageCount = 1, PageSize = 100 }),
+                var response = await RetryUtility.Retry(
+                    () => s_httpClient.SendAsync(
+                        CreateHttpRequest(
+                            new StringContent(JsonUtility.Serialize(request), System.Text.Encoding.UTF8, "application/json"))),
                     ex => ex is OperationCanceledException);
 
-                return (null, commits.Select(ToGitHubUser));
-            }
-            catch (NotFoundException)
-            {
-                // owner/repo doesn't exist or you don't have access to the repo
-                _unknownRepos.TryAdd((repoOwner, repoName));
-                return default;
-            }
-            catch (ApiValidationException)
-            {
-                // commit does not exist
-                return default;
+                if (!response.IsSuccessStatusCode)
+                {
+                    var message = await response.Content.ReadAsStringAsync();
+                    return (Errors.GitHubApiFailed(_url, message), null);
+                }
+
+                var grahpApiResponse = JsonConvert.DeserializeAnonymousType(await response.Content.ReadAsStringAsync(), new
+                {
+                    errors = new[]
+                    {
+                        new
+                        {
+                            message = "",
+                        },
+                    },
+                    data = new
+                    {
+                        repository = new
+                        {
+                            @object = new
+                            {
+                                history = new
+                                {
+                                    nodes = new[]
+                                    {
+                                        new
+                                        {
+                                            author = new
+                                            {
+                                                email = "",
+                                                user = new
+                                                {
+                                                    databaseId = 0,
+                                                    login = "",
+                                                    name = "",
+                                                },
+                                            },
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                });
+
+                if (grahpApiResponse.errors != null && grahpApiResponse.errors.Any())
+                {
```

Comment:
Before you are using HttpClient directly, the old `try catch` does not work, so handle these special cases here:
- when rate limit exceed, do not attempt to call GitHub again by setting `_rateLimitError`
- when the repo does not exist or the caller does have access, added it to `_unknownRepos` and do not attempt to call GitHub for this repo again.

Reply:
good point